run_gcs_to_bq/common/gcs_to_bq_util.py

```diff
@@ -0,0 +1,75 @@
+import os
+import json
+import time
+from google.cloud import bigquery, storage
+from pandas import DataFrame
+
+
+def append_dataframe_to_bq(frame, column_types, dataset, table_name):
+  """Appends the provided DataFrame to the table specified by
+     `dataset.table_name`. Automatically adds an ingestion time column.
+
+     frame: pandas.DataFrame representing the data to append.
+     column_types: A dict of column name to BigQuery data type. The column
+                   names must match the columns in the DataFrame
+     dataset: The BigQuery dataset to write to
+     table_name: The BigQuery table to write to"""
+  input_cols = column_types.keys()
+  if (len(input_cols) != len(frame.columns)
+          or set(input_cols) != set(frame.columns)):
+    raise Exception('Column types did not match frame columns')
+
+  columns = column_types.copy()
+  columns['ingestion_time'] = 'TIMESTAMP'
+  ingestion_time = time.time()
+  frame['ingestion_time'] = ingestion_time
+
+  client = bigquery.Client()
+  dataset_ref = client.dataset(dataset)
+  job_config = bigquery.LoadJobConfig()
+
+  # Always append, so we can keep the history.
```

Comment:
I know this has nothing to do with your change, but we should revisit how we do this for the main project. It would be neater to export the history to a separate archive.